packages/firestore/src/local/memory_remote_document_cache.ts

```diff
@@ -50,7 +50,18 @@ function documentEntryMap(): DocumentEntryMap {
   );
 }
 
-export class MemoryRemoteDocumentCache implements RemoteDocumentCache {
+export interface MemoryRemoteDocumentCache extends RemoteDocumentCache {
+  forEachDocumentKey(
+    transaction: PersistenceTransaction,
+    f: (key: DocumentKey) => PersistencePromise<void>
+  ): PersistencePromise<void>;
+}
+
+/**
+ * The memory-only RemoteDocumentCache for IndexedDb. To construct, invoke
+ * `newMemoryRemoteDocumentCache()`.
+ */
+class MemoryRemoteDocumentCacheImpl implements RemoteDocumentCache {
```

Comment:
Should `MemoryRemoteDocumentCacheImpl` implement `MemoryRemoteDocumentCache` instead of `RemoteDocumentCache`?

Reply:
Yes. Thanks for catching.